### ISSUE TITLE
new check: detect non-portable use of 'read' builtin with no variable name

### ISSUE THREAD

**Issue by pm215** [koalaman/shellcheck#2805]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

(This is mentioned in a comment on https://github.com/koalaman/shellcheck/issues/401 but I think this is a distinct feature request from the bug that issue is reporting.)

The 'read' builtin on bash allows you to omit the variable name, in which case it will set REPLY. However, this isn't portable, and dash will complain:
```
zz9.sh: 2: read: arg count
```
It would be nice if shellcheck detected this non-portability and suggested the use of 'read REPLY' instead.

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
read -r

```

#### Here's what shellcheck currently says:
```
No issues detected!
```
#### Here's what I wanted or expected to see:
```
Line 2:
read -r
^-- SCNNNN (error): read without a variable name is not portable
```
(or something similar).

**Comment by koalaman** [comment]
This has already been implemented in #2625, and will be part of the next release. Thanks, @ArenM!
